cognite/client/data_classes/data_modeling/containers.py

```diff
@@ -0,0 +1,314 @@
+from __future__ import annotations
+
+import json
+from abc import ABC
+from dataclasses import asdict, dataclass
+from typing import TYPE_CHECKING, Any, Literal, Optional, cast
+
+from cognite.client.data_classes._base import (
+    CogniteFilter,
+    CogniteResourceList,
+)
+from cognite.client.data_classes.data_modeling._validation import validate_data_modeling_identifier
+from cognite.client.data_classes.data_modeling.shared import (
+    CDFExternalIdReference,
+    ContainerReference,
+    DataModeling,
+    DirectNodeRelation,
+    Primitive,
+    PropertyType,
+    Text,
+)
+from cognite.client.utils._auxiliary import rename_and_exclude_keys
+from cognite.client.utils._text import convert_all_keys_to_camel_case_recursive, convert_all_keys_to_snake_case
+
+if TYPE_CHECKING:
+    from cognite.client import CogniteClient
+
+
+class ContainerCore(DataModeling):
+    """Represent the physical storage of data. This is the base class for the read and write version.
+
+    Args:
+        space (str): The workspace for the view, a unique identifier for the space.
+        external_id (str): Combined with the space is the unique identifier of the view.
+        description (str): Textual description of the view
+        name (str): Human readable name for the view.
+        used_for (Literal['node', 'edge', 'all']): Should this operation apply to nodes, edges or both.
+        properties (dict[str, ContainerPropertyIdentifier]): We index the property by a local unique identifier.
+        constraints (dict[str, ConstraintIdentifier]): Set of constraints to apply to the container
+        indexes (dict[str, IndexIdentifier]): Set of indexes to apply to the container.
+    """
+
+    def __init__(
+        self,
+        space: str = None,
+        external_id: str = None,
```

Comment:
These aren't optional
```suggestion
        space: str,
        external_id: str,
```